core/src/runtime/opaque.rs

```diff
@@ -25,6 +32,13 @@ pub(crate) struct Opaque<'js> {
     /// The user provided interrupt handler, if any.
     interrupt_handler: UnsafeCell<Option<InterruptHandler>>,
 
+    /// The class id for rust classes.
+    class_id: qjs::JSClassID,
+    /// The class id for rust classes which can be called.
+    callable_class_id: qjs::JSClassID,
+
+    prototypes: UnsafeCell<HashMap<*const VTable, Option<Object<'js>>>>,
```

Comment:
We can use trait top level recursion to create unique auto incrementing IDs at compile time. I think this could work and has zero cost
```rust
pub trait Counter {
    const NEXT: usize;
}

impl Counter for () {
    const NEXT: usize = 0;
}

//we don't care about T just want to increment the value
pub struct Increment<T>(std::marker::PhantomData<T>);

impl<T: Counter> Counter for Increment<T> {
    const NEXT: usize = T::NEXT + 1;
}
```

```rust
const VTABLE: VTable = {
  // generate unique ID for this VTable using type-level counter
  const ID: usize = <Increment<Increment<()>> as Counter>::NEXT;
  
  VTable {
      id: TypeId::of::<C::Changed<'static>>,
      vtable_id: ID,  // Assign unique compile-time ID
      finalizer: VTable::finalizer_impl::<'js, C>,
      trace: VTable::trace_impl::<C>,
      call: VTable::call_impl::<C>,
  }
};
```